packages/query-core/src/types.ts

```diff
@@ -27,7 +27,9 @@ export interface QueryFunctionContext<
 
 export type InitialDataFunction<T> = () => T | undefined
 
-export type PlaceholderDataFunction<TResult> = () => TResult | undefined
+export type PlaceholderDataFunction<TQueryFnData> = (
+  previousData: TQueryFnData | undefined,
+) => TQueryFnData | undefined
```

Comment:
to be consistent, I think the generic should be named `TQueryData` here. It's what we use at the call-side. The only difference is that `TQueryFn` data is _exactly_ what the `queryFn` returns, while `TQueryData` might be a different structure - mainly for infinite queries. For normal queries, those two are identical.

Reply:
✅